docs/data/material/components/material-icons/SearchIcons.js

```diff
@@ -149,21 +155,49 @@ function handleLabelClick(event) {
 
 function Icon(props) {
   const { icon, onOpenClick } = props;
+
+  const rootRef = React.useRef(null);
+  const [isVisible, setIsVisible] = React.useState(false);
+
+  React.useEffect(() => {
+    const margin = 200;
+    const observer = new IntersectionObserver(
+      (entries) => {
+        // `isIntersecting` is not working correctly on Chrome
+        const clientRect = entries[0].target.getBoundingClientRect();
+        const isIntersecting =
+          clientRect.top < window.innerHeight + margin &&
+          clientRect.bottom > -margin;
+        setIsVisible(isIntersecting);
```

Comment:
It doesn't trigger for me on the initial load on mobile.

Reply:
Should we ever unmount SVG icons? I mean once it's on the page, is there a cost?